### ISSUE TITLE
Warning of experimental support for decorators

### ISSUE THREAD

**Issue by philjones88** [AngularClass/lib-starter#4]
severity: 'Error'
message: 'Experimental support for decorators is a feature that is subject to change in a future release. Set the 'experimentalDecorators' option to remove this warning.'

**Comment by Hendrixer** [comment]
are you getting this error in VS code? or build time? I believe it has to do with the `files` array in the tsconfig and not putting in a `rootDir` in there are well.

**Comment by philjones88** [comment]
In VS Code (insiders).

I don't get this on other projects where I used your `angular-starter` project so maybe it's a difference in the tsconfig files? 

https://github.com/AngularClass/angular-starter/blob/master/tsconfig.json

**Comment by Hendrixer** [comment]
right, i can confirm it has to do with the lack of `"rootDir"` in the tsconfig

**Comment by philjones88** [comment]
I don't know if that's right? I use the angular-starter tsconfig.json in our product and we don't get this warning? Let me test what settings from the angular-starter tsconfig.json fix it :)

**Comment by Hendrixer** [comment]
in my case it was. The real issue is when you have a file not being included by the tsconfig. This can be from rootDir, excludes, or files.

**Comment by philjones88** [comment]
I can't replicate it tonight. VS Code has now decided that in `lib.component.ts` that the `template` property is missing even though it's right there! :(

```
file: 'file:///Users/phil/Documents/Projects/lib-starter/src/lib.component.ts'
severity: 'Error'
message: 'Component LibComponent must have a template or templateUrl'
at: '3,2'
source: 'Angular'
```

**Comment by AlirezaEbrahimkhani** [comment]
In VSCode, Go to File => Preferences => Settings (or Control+comma) and it will open the User Settings file. Add "javascript.implicitProjectConfig.experimentalDecorators": true to the file and it should fix it. It did for me.
